1. Make an API request that returns the months where "fire" was searched in 2016. Which month had the most searches?  

In [51]:
import requests
import pandas as pd

In [17]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json'
params = {
    'year': '2016',
    'query_text': 'fire'
}
response = requests.get(endpoint, params = params)
response

<Response [200]>

In [31]:
res = response.json()
res

[{'month_name': 'January',
  'year': '2016',
  'query_count': '19',
  'query_text': 'fire'},
 {'month_name': 'February',
  'year': '2016',
  'query_count': '35',
  'query_text': 'fire'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '32',
  'query_text': 'fire'},
 {'month_name': 'April',
  'year': '2016',
  'query_count': '26',
  'query_text': 'fire'},
 {'month_name': 'May',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'June',
  'year': '2016',
  'query_count': '31',
  'query_text': 'fire'},
 {'month_name': 'July',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'August',
  'year': '2016',
  'query_count': '47',
  'query_text': 'fire'},
 {'month_name': 'September',
  'year': '2016',
  'query_count': '36',
  'query_text': 'fire'},
 {'month_name': 'October',
  'year': '2016',
  'query_count': '38',
  'query_text': 'fire'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '32',
  'query_text

In [97]:
fire_searches = pd.DataFrame({'Month': [res[x].get('month_name') for x in range(0, 12)], 'Searches': [res[x].get('query_count') for x in range(0, 12)]})
fire_searches['Month'] = fire_searches['Month'].astype('string')
fire_searches['Searches'] = fire_searches['Searches'].astype('int')
fire_searches


Month  Searches
0     January        19
1    February        35
2       March        32
3       April        26
4         May        24
5        June        31
6        July        24
7      August        47
8   September        36
9     October        38
10   November        32
11   December        38

In [101]:
total = fire_searches['Searches'].sum()
total

382

In [105]:
highest_vol = fire_searches[fire_searches['Searches']]
highest_vol

KeyError: 47

There were 382 searches for "fire" in 2016, with August being the highest, at 47.